compiler/modules/bir/verify.bal

```diff
@@ -298,6 +300,17 @@ function verifyCompare(VerifyContext vc, CompareInsn insn) returns err:Semantic?
             operandType = t:constBasicType(operand);
         }
         if !t:isSubtypeSimple(operandType, expectType) {
+            if memberType is t:UniformTypeBitSet {
```

Comment:
You should factor out the checking for the non-array case and reuse for checking the member of the array.

Reply:
fixed